features/netsocket/nsapi_types.h

```diff
@@ -190,6 +190,37 @@ typedef struct nsapi_addr {
      *  The raw bytes of the IP address stored in big-endian format
      */
     uint8_t bytes[NSAPI_IP_BYTES];
+
+    /** Default constructor */
+    nsapi_addr() :
```

Comment:
If you recompile, you might notice that `nsapi_types.h` is also included in C files. So this needs to be pure C99.

Reply:
Thanks for pointing this out, @SeppoTakalo .
Any idea how to make sure a C structure will always be initialized? I was trying to avoid initializing every instance of nsapi_addr we have in our code...